### PR TITLE
Refactor code to follow data handling case conventions

### LINKED ISSUE #102: Ensure correct case when handling data for all commands
Ensure the following for all commands:

Display and Storage:
- matricNumber --> convert to UPPERCASE
- Other data --> keep as original case

Matching and Finding:
- Set to UPPERCASE before comparison

### PR DESCRIPTION
**Pull request by yeekian**
Fixes #102 

Added author tags to preserve code ownership of affected methods.

### CODE REVIEW

File: src/main/java/tutorlink/command/AddGradeCommand.java [hunk 1]
```diff
@@ -15,11 +15,13 @@
 import tutorlink.result.CommandResult;
 import tutorlink.student.Student;
 import java.util.HashMap;
+
 import static tutorlink.lists.StudentList.STUDENT_NOT_FOUND;
 
 public class AddGradeCommand extends Command {
     public static final String[] ARGUMENT_PREFIXES = {"i/", "c/", "s/"};
     public static final String COMMAND_WORD = "add_grade";
+    private static final String ERROR_DUPLICATE_GRADE_ON_ADD = "Error! Grade (%s, %s) already exists in the list!";
```
Comment (RCPilot1604) [inline src/main/java/tutorlink/command/AddGradeCommand.java:24]: Should this be extracted out in Commons?

**Review by RCPilot1604 — APPROVED** [review]
Looks good to me. 

There are quite a number of command messages that are not extracted out in Commons. However in terms of functionality it is good to go.

File: src/main/java/tutorlink/command/AddGradeCommand.java (hunk 1 above)
Comment (yeekian) [reply src/main/java/tutorlink/command/AddGradeCommand.java:24]: Sure, I'll make a new issue for us to check for all. #112

File: src/test/java/tutorlink/lists/ComponentListTest.java
Comment (jinzihan2002) [inline src/test/java/tutorlink/lists/ComponentListTest.java:1]: Thanks for updating the tests to continue functioning with the updated components!

**Review by jinzihan2002 — APPROVED** [review]
LGTM

**Pull request merged**